packages/core/admin/admin/src/utils/getFetchClient.js

```diff
@@ -0,0 +1,48 @@
+import axios from 'axios';
+import { auth } from '@strapi/helper-plugin';
+
+// eslint-disable-next-line import/no-mutable-exports
+export let instance;
```

Comment:
Yes, our first option was to do that, it seemed unnecessary to us to pass the Axios instance through the provider when it doesn’t really have a state. Instead, we could just create a singleton function like this and call it through the hook directly. How do you see this? https://www.notion.so/strapi/ADRs-Architecture-Decision-Records-b5b9042c38f7449088cbaba540904361?p=a7d01bde64d44b1c84570983923dbc7b&pm=s

Reply:
I would, why are we adding additional unnecessary complexity via out context providers, they're already tricky to manage.

Reply:
@alexandrebodin could you approve this?

Reply:
The only reason to pass it via providers IMO is if it needs to be initialized at the StrapiApp level. No problem doing this in the hook instead if that's sufficient 👍 

That being said looking at the code it looks like you don't even need a hook to access this 🤔  a simple import of the instance would work here. 

FYI reading this code I don't really see the point of using this singleton pattern like this. it makes only sense when you need to pass arguments for the 1st initialization or something. 

Here you can just define the instance in the module and export it as modules are singletons by nature.


```js
// export it for testing purposes and pass dependencies as params for testing too
export const createFetchClient = ({ baseURL }) => {
  return axios.create();
}

export default createFetchClient({ baseURL: process.env.STRAPI_ADMIN_BACKEND_URL })
```

It looks like you could just do  `import fechClient from '@strapi/admin'`

Reply:
> Here you can just define the instance in the module and export it as modules are singletons by nature.

I like this. The _only_ reason I can think of writing a `useFetchClient` hook would be if we are going to implement a cancelling mechanism for if the component is unmounted before the `axios` request has finished. Whereby we can cancel calls when the hook is unmounted. Because the hook returns and adds `signal` to the calls only those calls made by the specific hook would be cancelled as opposed to all the calls the instance wants to make.

There are other scenarios we could manage e.g. avoiding race conditions on multiple requests but that might be out of scope for now and may be something axios handles already.

Reply:
That sounds good. We can mix the to and expose the imported instance through a hook to avoid breaking the api later on in that case.

Reply:
Hello, thanks for your suggestions. The main idea before the implementation of this axios migration was to avoid duplicated implementations of the same utility. In addiction, we tried to avoid exposing it inside the helper-plugin because we know that its presence in the future is under discussion. I think these are the reasons to use the StrapiApp Context to pass the utility and to expose, in the beginning, the hook inside the helper plugin. Now your suggested solution means removing the getFetchClient from the StrapiApp, and having the implementation of the getFetchClient code inside the Admin and the hook inside the helper-plugin, I can also try to move the implementation of the getFetchClient code in the helper-plugin but in this case we are against our first idea, to add more things in the helper-plugin, wdyt?

Reply:
I'm not sure I see why we can't still do all this in `@strapi/admin` to plan for the helper plugin deprecation ?

Reply:
yes I moved all the logic in the admin now

Reply:
it is ready to be checked @joshuaellis and @alexandrebodin